### PR TITLE
Optimization of TreegraphSeries

### PR DESCRIPTION
**Pull request by bre1470**
- Replaced let and forEach.
- Fixed doclets.

### CODE REVIEW

File: ts/Series/Treegraph/TreegraphSeriesDefaults.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ import { Palette } from '../../Core/Color/Palettes';
  *
  * TODO change back the demo path
  * @sample highcharts/demo/treegraph-chart
- *          Treegraph Chart
+ *         Treegraph Chart
```
Comment (bre1470) [inline ts/Series/Treegraph/TreegraphSeriesDefaults.ts:38]: @pawellysy There is a todo about the missing sample. I do not see how this needs to be changed back. Can you take a look, please?

Comment (pawellysy) [reply ts/Series/Treegraph/TreegraphSeriesDefaults.ts:38]: We agreed with @TorsteinHonsi, that Treegraph won't be released until we change the design, so I changed all demos, to not be referenced in the official docs. More info here: #17600

File: ts/Series/Treegraph/TreegraphSeriesDefaults.ts [hunk 2]
```diff
@@ -22,20 +22,19 @@ import type TreegraphSeriesOptions from './TreegraphSeriesOptions';
...
 
 /**
  * A treegraph series is a diagram, which shows a relation between ancestors
  * and descendants with a clear parent - child relation.
  * The best examples of the dataStructures, which best reflect this chart
  * are e.g. genealogy tree or directory structure.
  *
- * TODO change back the demo path
- * @sample highcharts/demo/treegraph-chart
- *          Treegraph Chart
```
Comment (pawellysy) [inline ts/Series/Treegraph/TreegraphSeriesDefaults.ts:38]: @bre1470  the idea was to change the sample's path back when PR #17600 gets merged. (apart from the fact, that there is a typo `highchart`, was this intended?)

Comment (bre1470) [reply ts/Series/Treegraph/TreegraphSeriesDefaults.ts]: I think it is best if you fix the issue with treegraph samples. The warnings about invalid `@sample` will show when `npm run jsdoc` on master.

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**